File: app/src/main/java/com/androidexamples/app/activity/AnimationsActivity.java
package com.androidexamples.app.activity;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.animation.AnimationSet;

import com.androidexamples.app.R;

public class AnimationsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_animations);
        loadComponents();
    }

    private void loadComponents() {
        setSupportActionBar((Toolbar)findViewById(R.id.toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        findViewById(R.id.button_view_animations).setOnClickListener(onButtonClick());
        findViewById(R.id.button_activity_animations).setOnClickListener(onButtonClick());
        findViewById(R.id.button_reveal_effect).setOnClickListener(onButtonClick());
    }

    private View.OnClickListener onButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                switch (v.getId()) {
                    case R.id.button_view_animations:
                        startActivity(new Intent(getBaseContext(), ViewAnimationsActivity.class));
                        break;
                    case R.id.button_activity_animations:
                        startActivity(new Intent(getBaseContext(), ActivityAnimationsActivity.class));
                        break;
                    case R.id.button_reveal_effect:
                        startActivity(new Intent(getBaseContext(), RevealEffectActivity.class));
                        break;
                }
            }
        };
    }
}

File: app/src/main/java/com/androidexamples/app/flows/animations/activityAnimations/AnimationsActivity.kt
package com.androidexamples.app.flows.animations.activityAnimations

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.View

import com.androidexamples.app.R
import com.androidexamples.app.flows.animations.viewAnimations.ViewAnimationsActivity

class AnimationsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_animations)
        loadComponents()
    }

    private fun loadComponents() {
        setSupportActionBar(findViewById<View>(R.id.top_toolbar) as Toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        findViewById<View>(R.id.button_view_animations).setOnClickListener(onButtonClick())
        findViewById<View>(R.id.button_activity_animations).setOnClickListener(onButtonClick())
        findViewById<View>(R.id.button_reveal_effect).setOnClickListener(onButtonClick())
    }

    private fun onButtonClick(): View.OnClickListener {
        return View.OnClickListener { v ->
            when (v.id) {
                R.id.button_view_animations -> startActivity(Intent(baseContext, ViewAnimationsActivity::class.java))
                R.id.button_activity_animations -> startActivity(Intent(baseContext, ActivityAnimationsActivity::class.java))
                R.id.button_reveal_effect -> startActivity(Intent(baseContext, RevealEffectActivity::class.java))
            }
        }
    }
}
